swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+        other_vendor.inventory.remove(their_item)
+        return True 
+        
+    def swap_first_item(self, other_vendor):
+        if not (self.inventory and other_vendor.inventory):
+            return False 
+        my_first_item = self.inventory[0]
+        other_vendor_first_item = other_vendor.inventory[0]
+        self.swap_items(other_vendor, my_first_item, other_vendor_first_item)       
+        return True 
+    
+    def get_by_category(self, category):
+        inventory_by_category = [item for item in self.inventory if category == item.get_category()]
+        return inventory_by_category
+    
+    #Returns item with highest value for condition attribute. 
+    def get_best_by_category(self, category):
+        inventory_by_category = self.get_by_category(category)
+        if not inventory_by_category:
+            return None 
+        item_best_condition = max(inventory_by_category, key=lambda item: item.condition)
+        return item_best_condition   
+    
+    #Returns true if item with highest value for condition is swapped between vendors.
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        if not (self.get_by_category(their_priority) and other_vendor.get_by_category(my_priority)):
+            return False 
+        
+        vendor_best_item = self.get_best_by_category(their_priority)
+        other_vendor_best_item = other_vendor.get_best_by_category(my_priority)
+        self.swap_items(other_vendor, vendor_best_item, other_vendor.get_best_by_category(my_priority))
+        return True 
+    
+    def get_newest_by_category(self, category):
+        inventory_by_category = self.get_by_category(category)
+        if not inventory_by_category:
+            return None 
+        item_newest = min(inventory_by_category, key=lambda item: item.age)
+        return item_newest
+    
+    def swap_newest_by_category(self, other_vendor, my_priority, their_priority):
+        if not (self.get_by_category(their_priority) and other_vendor.get_by_category(my_priority)):
+            return False 
+        
+        vendor_newest_item = self.get_newest_by_category(their_priority)
+        other_vendor_newest_item = other_vendor.get_newest_by_category(my_priority)
+        self.swap_items(other_vendor, vendor_newest_item, other_vendor_newest_item)
+        return True 
+    
+    #This is a possible higher order function to combine swap by age or condition and category.
+    #It is commented out to not change test_wave_06
+    #
+    # def swap_modifier_by_category(self, other_vendor, my_priority, their_priority, swap_modifier):
+    #     if not (self.get_by_category(their_priority) and other_vendor.get_by_category(my_priority)):
+    #         return False 
+    #     vendor_item = swap_modifier(self, their_priority)
+    #     other_vendor_item = swap_modifier(other_vendor, my_priority)
+    #     self.swap_items(other_vendor, vendor_item, other_vendor_item)
```

Comment:
This is FANTASTIC! I'm so glad you got to practice higher-order functions. I love this, great work, keep this up!!